File: MagTech.Models/BazaArtykulow.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MagTech.Models
{
    public class BazaArtykulow
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [MaxLength(19)]
        public string NumerArtykulu { get; set; }
        [Required]      
        public int NumerRegalu { get; set; }
        [Required]      
        public int NumerPrzedzialu { get; set; }
        [Required]      
        public int NumerPolki { get; set; }
        [Required]      
        public int StanWMiejscuSkladowania { get; set; }
        [Required]       
        public int OznaczenieFifo { get; set; }
        [MaxLength(5)]
        public string WielkoscPojemnika { get; set; }       
        public int StanMinimalny { get; set; }       
        public int NumerSzeregu { get; set; }
        [MaxLength(40)]
        public string NazwaArtykulu { get; set; }
        [MaxLength(10)]
        public string GrupaMaterialowa { get; set; }
        [MaxLength(2)]
        public string Typ { get; set; }
        [MaxLength(10)]
        public string Tag { get; set; }
        [MaxLength(120)]
        public string DodatkoweInformacje { get; set; }
    }
}

File: MagTech/wwwroot/js/bazaartykulow.js
var dataTable;

$(document).ready(function () {
    $('#tblData tfoot th').each(function () {
        var title = $(this).text();
        $(this).html('<input type="text" placeholder="Szukaj ' + title + '" />');
    });
    loadDataTable();       
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({   
        //"scrollX": true,       
        "ajax": {
            "url": "/Admin/BazaArtykulow/GetAll"
        },        
        "columns": [                
            { "data": "numerArtykulu", "width": "10%" },
            { "data": "nazwaArtykulu", "width": "15%"},
            { "data": "stanWMiejscuSkladowania", "width": "5%"},
            { "data": "stanMinimalny", "width": "5%" },
            { "data": "dodatkoweInformacje", "width": "10%" },
            { "data": "grupaMaterialowa", "width": "5%"},
            { "data": "numerRegalu", "width": "5%" },
            { "data": "numerPolki", "width": "5%"},
            { "data": "numerPrzedzialu", "width": "5%"},
            { "data": "numerSzeregu", "width": "5%" },
            { "data": "wielkoscPojemnika", "width": "5%" },
            { "data": "oznaczenieFifo", "width": "5%" },
            { "data": "typ", "width": "5%" },
            { "data": "tag", "width": "5%" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                            <div class="text-center">
                                <a href="/Admin/BazaArtykulow/Edit/${data}" class="btn btn-info" style="cursor:pointer" data-toggle="tooltip" data-placement="bottom" title="" data-original-title="Edytuj">
                                    <img src="/icons/edit-button.png" width="16" height="16" />
                                </a>                                
                            </div>
                           `;
                }, "width": "10%"
            }          
                     
        ],
        "language": {
            "lengthMenu": "Ilość _MENU_ rekordów na stronę",
            "zeroRecords": "Ładowanie danych",
            "info": "Strona _PAGE_ z _PAGES_",
            "infoEmpty": "Brak wyników",
            "infoFiltered": "(filtered from _MAX_ total records)",
            "search": "szukaj",
            "paginate": {
                "first": "Pierwsza",
                "previous": "Poprzedni",
                "next": "Następny",
                "last": "Ostatnia"
            },
        },
        // DataTable
            initComplete: function () {
                // Apply the search
                this.api().columns().every(function () {
                    var that = this;

                    $('input', this.footer()).on('keyup change clear', function () {
                        if (that.search() !== this.value) {
                            that
                                .search(this.value)
                                .draw();
                        }
                    });
                });
            }        
    });
}

